src/ddl/migrations/covid_hosp_facility_v0.3-v0.3.1.sql

```diff
@@ -0,0 +1,22 @@
+INSERT INTO `covid_hosp_facility_key`
+  (`address`,`ccn`,`city`,`fips_code`,`geocoded_hospital_address`,`hhs_ids`,`hospital_name`,`hospital_pk`,`hospital_subtype`,`is_metro_micro`,`state`,`zip`)
+SELECT
+   `address`,`ccn`,`city`,`fips_code`,`geocoded_hospital_address`,`hhs_ids`,`hospital_name`,`hospital_pk`,`hospital_subtype`,`is_metro_micro`,`state`,`zip`
+FROM `covid_hosp_facility` ORDER BY `id` DESC
+ON DUPLICATE KEY UPDATE hospital_pk=covid_hosp_facility_key.hospital_pk;
+
+-- -- considered this approach too (note opposite direction in the ORDER BY clause) :
+-- INSERT INTO `covid_hosp_facility_key`
+--   (`address`,`ccn`,`city`,`fips_code`,`geocoded_hospital_address`,`hhs_ids`,`hospital_name`,`hospital_pk`,`hospital_subtype`,`is_metro_micro`,`state`,`zip`)
+-- SELECT
+--    `address`,`ccn`,`city`,`fips_code`,`geocoded_hospital_address`,`hhs_ids`,`hospital_name`,`hospital_pk`,`hospital_subtype`,`is_metro_micro`,`state`,`zip`
+-- FROM `covid_hosp_facility` ORDER BY `id` ASC
+-- ON DUPLICATE KEY UPDATE `address`=`address`,`ccn`=`ccn`,`city`=`city`,`fips_code`=`fips_code`,`geocoded_hospital_address`=`geocoded_hospital_address`,`hhs_ids`=`hhs_ids`,`hospital_name`=`hospital_name`,`hospital_subtype`=`hospital_subtype`,`is_metro_micro`=`is_metro_micro`,`state`=`state`,`zip`=`zip`;
+
+-- -- or a query similar to the previous behavior of the covid_hosp_facility endpoint:
+-- INSERT INTO `covid_hosp_facility_key`
+--   (`hospital_pk`,`address`,`ccn`,`city`,`fips_code`,`geocoded_hospital_address`,`hhs_ids`,`hospital_name`,`hospital_subtype`,`is_metro_micro`,`state`,`zip`)
+-- SELECT
+--    `hospital_pk`,MAX(`address`),MAX(`ccn`),MAX(`city`),MAX(`fips_code`),MAX(`geocoded_hospital_address`),MAX(`hhs_ids`),MAX(`hospital_name`),MAX(`hospital_subtype`),MAX(`is_metro_micro`),MAX(`state`),MAX(`zip`)
+-- FROM `covid_hosp_facility`
+-- GROUP BY `hospital_pk`;
```

Comment:
All of the above methods assume we will fix https://github.com/cmu-delphi/delphi-epidata/issues/1007 before migration -- is that strictly necessary, or could we effect a migration by creating the table and then re-importing the most recent facility update?

Reply:
we can fix before migration, or we can TRUNCATE and re-run this INSERT/SELECT as part of the fix (after the other table has been rectified).  we could also remove bad rows from this table as part of the fix, but that requires identifying them which may or may not be easy depending on how the fix is implemented.